File: java/JavaDemo/src/main/java/com/bandwidth/enviroment/Properties.java
package com.bandwidth.enviroment;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import static java.lang.System.out;

public class Properties {


    private static final Map<String, String> PROPS = new LinkedHashMap<>();

    private Properties(){

    }

    public static void configure(String filePath){

        File file = new File(filePath);

        if(!file.exists())
            return;


        List<String> logList = null;
        try {
            logList = Files.readAllLines(file.toPath());
        } catch (IOException e) {
            out.println(e.getMessage());
        }


        for(String line : logList){
            int index = line.indexOf('=');
            if(index == -1 || index >= line.length()) continue;

            String left = line.substring(0, index);
            String right = line.substring(index + 1, line.length() );

            PROPS.put(left, right);
        }
    }

    public static String getProperty(String key){

        if(PROPS.containsKey(key)){
            return PROPS.get(key);
        } else {
            return null;
        }


    }
}

File: csharp/BandwidthReferenceAPp/Enviroment/Properties.cs
using System.IO;
using System.Collections.Generic;
namespace Enviroment {

	public class Properties {

		private static readonly Dictionary<string, string> PROPS = new Dictionary<string, string>();

		private Properties(){

		}

		public static void configure(string filePath){

			if(!File.Exists(filePath))
				return;
			

			string[] logFile = File.ReadAllLines(filePath);
			List<string> logList = new List<string>(logFile);


			foreach(string line in logList){
				int index = line.IndexOf('=');
				if(index == -1 || index >= line.Length) continue;

				string left = line.Substring(0, index);
				string right = line.Substring(index + 1, line.Length - index - 1);

				PROPS[left] = right;
			}
		}

		public static string getProperty(string key){

			if(PROPS.ContainsKey(key)){
				return PROPS[key];
			} else {
				return null;
			}
			
		}
	}
}